# Juliaで学ぶタイトバインディング模型とトポロジカル物質
## 2. タイトバインディング模型と固体中の電子
前回のノートでは、一次元シュレーディンガー方程式から始めてタイトバインディング模型を導入した。この模型を解くことで固体中の電子の挙動をいろいろと理解することができる。しかしながら、一つのことを考えなければならない。「固体中には無数の電子がいるが、これをどのように扱えばよいか？」である。

このノートでは、多体電子系のシュレーディンガー方程式について調べ、密度汎関数理論の導入によってタイトバインディング模型に正当性が現れることを示す。最後に、タイトバインディング模型の性質に軽く触れる。


### 2.1 多体電子系のシュレーディンガー方程式
これまではずっと一粒子のシュレーディンガー方程式を考えてきた。しかし、固体中には無数の電子がいる。このとき、電子集団が従う方程式は多体のシュレーディンガー方程式となる。多体のシュレーディンガー方程式は
$$
H(r_1,r_2,\cdots,r_N) \psi(r_1,r_2,\cdots,r_N) = E \psi(r_1,r_2,\cdots,r_N) 
$$
と書くことができる。これだとよくわからないので、$N=2$の場合を考える。この時、
$$
\left( -\frac{\hbar^2}{2m} \frac{\partial^2}{\partial r_1^2} -\frac{\hbar^2}{2m} \frac{\partial^2}{\partial r_2^2} + V(r_1) + V(r_2) + U(r_1,r_2) \right)  \psi(r_1,r_2) = E \psi(r_1,r_2) 
$$
と書ける。ここで、$V(r_1)$は座標を$r_1$でラベルされた電子の感じるポテンシャル、$U(r_1,r_2)$はクーロン相互作用などの電子間相互作用である。  
さて、電子間相互作用をひとまず無視:
$$
\left( -\frac{\hbar^2}{2m} \frac{\partial^2}{\partial r_1^2} -\frac{\hbar^2}{2m} \frac{\partial^2}{\partial r_2^2} + V(r_1) + V(r_2)  \right)  \psi(r_1,r_2) = E \psi(r_1,r_2) 
$$
してみよう。このとき、一電子に関するシュレーディンガー方程式の解を
$$
\left( -\frac{\hbar^2}{2m} \frac{\partial^2}{\partial r_1^2} + V(r_1) \right) \phi_{n1}(r_1) = \epsilon_{n1} \phi_{n1}(r_1)
$$
$$
\left( -\frac{\hbar^2}{2m} \frac{\partial^2}{\partial r_2^2} + V(r_2) \right) \phi_{n2}(r_2) = \epsilon_{n2} \phi_{n2}(r_2)
$$
と導入すると、
$$
\psi(r_1,r_2) = \phi_{n1}(r_1) \phi_{n2}(r_2)
$$
は明らかに二体のシュレーディンガー方程式の解となっており、エネルギーは
$$
E = \epsilon_{n1}+\epsilon_{n2}
$$
である。エネルギーが最小となるのは、$\epsilon_{n1}=\epsilon_{n2} = \epsilon_0$、つまり同じ波動関数であるときである。しかし、後述するように、電子系ではこの波動関数をとることができない。

### 2.2 フェルミ粒子性
量子力学においては、粒子はお互いに区別できないことを思い出そう。つまり、電子Aと電子Bの違いは区別できない。つまり、電子Aが$r_1$にいて電子Bが$r_2$にいる状態と、電子Bが$r_1$にいて電子Aが$r_2$にいる状態は区別できない。
一方、波動関数は複素数なので、位相の自由度はついてもよい。つまり、
$$
\psi(r_2,r_1) = e^{i \theta} \psi(r_1,r_2)
$$
でなければならない。一方、再び$r_1$と$r_2$を入れ替えたときにもとに戻ってこないといけないので、
$$
\psi(r_1,r_2) = e^{i 2 \theta} \psi(r_1,r_2) = \psi(r_1,r_2)
$$
となり、$\theta = \pi$か$\theta = 0$でなければならない。$\theta = 0$のときはボソン（ボース粒子）、$\theta = \pi$のときはフェルミオン（フェルミ粒子）、と呼ばれている。電子はフェルミオンなので、$\theta = \pi$であり、
$$
\psi(r_2,r_1) = - \psi(r_1,r_2)
$$
である。このような関数を反対称関数と呼ぶ。
一方、多体のシュレーディンガー方程式の解$\psi(r_1,r_2) = \phi_{n1}(r_1) \phi_{n2}(r_2)$は、この条件を満たしていない。特に、両方とも同じエネルギーをもつ状態:
$$
\psi(r_1,r_2) = \phi_{n}(r_1)\phi_n(r_2)
$$
は反対称関数ではなく、禁じられている。次に低いエネルギーを持つ状態は二つあり、
$$
\phi_{0}(r_1)\phi_1(r_2), \phi_0(r_2)\phi_{1}(r_1)
$$
エネルギーは$\epsilon_0 + \epsilon_1$で縮退している。この二つの波動関数は両方とも二体のシュレーディンガー方程式の解なので、これらの線形結合もやはり解である。線形結合の中で、反対称関数となるものは
$$
\psi(r_1,r_2) = \phi_{0}(r_1)\phi_1(r_2) -  \phi_0(r_2)\phi_{1}(r_1)
$$
である。  
いま、電子を2個だけ考えたが、固体中には電子はもっと沢山いる。
もし、電子同士の相互作用がなければ、$N$電子の波動関数は
$$
\psi(r_1,\cdots,r_N) = \phi_1(r_1)\cdots \phi_N(r_N)
$$
はシュレーディンガー方程式の解である。ここで、$\phi_n(r)$は固有値$\epsilon_n$を持つ一体のシュレーディンガー方程式の固有関数であり、$\epsilon_1 < \cdots < \epsilon_N$とする。この時、全エネルギーは
$$
E = \sum_i \epsilon_i
$$
とすると最小エネルギーになる。なぜならば、フェルミオンであるために、同じエネルギーを持つことができないからである。
しかし、これではまだ、座標の入れ替えに対する反対称関数にはなっていないため、電子の状態を表していない。２個の時と同じように考えるならば、座標の足を入れ替えた関数を用意して、適切に符号を考えて足し上げてあげればよいはずである。全部で$N$個の座標があり、それがどのような組み合わせになっているかを表す記号を$C_i$とおく。$i$は$i$番目に考えている組み合わせである。このとき、$C_i$のラベルを持つ波動関数は
$$
\psi_{C_i} =  \phi_1(r_{C_i(1)})\cdots \phi_N(r_{C_i(N)})
$$
である。例えば２個の場合には、組み合わせは$C_1 = (r_1,r_2)$と$C_2 = (r_2,r_1)$となる。ここで、入れ替えたときに辻褄があうように導入する符号を${\rm sgn} (C)$とすると、シュレーディンガー方程式を満たし、かつ、フェルミオンの反対称性を満たすのは、
$$
\Psi(r_1,\cdots,r_N) = \sum_C {\rm sgn}\rm C \psi_1(r_{C(1)}) \cdots \psi_N(r_{C(N)})
$$
と書ける。  
電子同士の相互作用を考えていないにも関わらず、波動関数は大分ややこしくみえる。とくに、符号をどのようにすればよいのか、
わかりにくい。
しかし、線形代数を思い出せば、この式は実はわかりやすい形にかけていることがわかる。${\rm sgm} \rm C$は置換と呼ばれるもので、
「行列式」の定義そのままになっている。  
https://ja.wikipedia.org/wiki/行列式  
つまり、行列として、
$$
\hat{\psi} = 
\left( \begin{matrix}
\phi_1(r_1) & \cdots & \phi_1(r_N) \\
\vdots & & \vdots \\
\phi_N(r_1) & \cdots & \phi_N(r_N)
\end{matrix}
\right)
$$
を考えれば、波動関数は
$$
\Psi(r_1,\cdots,r_N) = \det \hat{\psi}
$$
と書けてしまうのである。  
この行列式をSlator行列式と呼ぶ。行列式でかけているので、$r_i$と$r_j$の座標の入れ替えは行列の縦を入れ替えることに対応しており、
行列式なのでもちろんマイナスの符号が出る。

以上より、相互作用がない系であれば、一粒子シュレーディンガー方程式を解き、エネルギーの低い順に$N$個固有関数を持ってくれば、多体のシュレーディンガー方程式の最低エネルギー状態を作ることができる。  
しかし、相互作用を無視するということは電子同士のクーロン相互作用などを無視しているように見える。これはずいぶん単純化された近似に見える。このような一体の方程式は役に立つのであろうか？

### 2.3 密度汎関数理論
相互作用のない方程式であれば、前回のノートのような方法でタイトバインディング模型を構築することができる。その結果、数百以上の原子が並んだ系を考えることができ、トポロジカル物質の端状態の議論もできるようになる。そして、相互作用がなければ、$N$粒子系のシュレーディンガー方程式は一粒子のシュレーディンガー方程式を解くことで解を作ることができる。つまり、一粒子のタイトバインディング模型を解くことで議論ができる。一方、相互作用が入ったシュレーディンガー方程式の解は一般に非常に複雑であり、座標の足が$N$個あるために$3N$次元中のシュレーディンガー方程式を解かなければ解を得ることができない。1次元のシュレーディンガー方程式を$m$個の離散座標に置き換えて差分化すると$m$次元行列の問題となる。$3N$次元のシュレーディンガー方程式であれば、$m^{3N}$次元という莫大な大きさの行列を解く必要がある。これを使うしか方法がなければ、固体物理の性質を理解することは非常に困難であった。  

実は、非常に巧妙な方法が存在する。それは「密度汎関数理論」である。  
https://ja.wikipedia.org/wiki/密度汎関数理論   
この理論では、ホーヘンベルク・コーンの定理と呼ばれるものを証明している。この定理は
「$N$粒子系の基底状態を表す3N次元の波動関数によって表現される基底状態の性質は、３つの空間座標に依存する、つまり３次元の電子密度だけで一意に定まる」である。以下にWikipediaの記事の一部分を引用しよう。
>ある外部ポテンシャルのもとにあるN個の電子系を考える（例えば分子の原子核の配置が決まれば、それらの原子核が電子に及ぼす静電ポテンシャルは決まる）。いま、この系の基底状態の電子密度ρだけがわかっているとする。ホーヘンベルク・コーンの第1定理によれば、ある系の基底状態の電子密度ρが決まると、それを基底状態にもつ外部ポテンシャルがもし存在すればそれはただ1通りに定まる。また電子数Nも電子密度を全空間に渡って積分することで求めることができる。その外部ポテンシャルと電子数から導かれるハミルトニアンHのシュレーディンガー方程式を解けば、その外部ポテンシャルのもとで許される電子系の波動関数Ψがわかるので、あらゆる物理量をそこから求めることができる。つまり、基底状態の電子密度から、系の（励起状態に関わる量も含めて）あらゆる物理量は原理的には計算できることになる。

ある相互作用のない系の方程式
$$
\left( -\frac{\hbar^2}{2m} \nabla^2 + V_{\rm eff}(r) \right) \psi_i(r) = \epsilon_i \psi_i(r)
$$
を考え、この系の基底状態の電子密度が実際の系の電子密度に一致するような有効ポテンシャル$V_{\rm eff}$を構成する理論のことを、コーンシャム理論と呼ぶ。
このとき、有効ポテンシャルはエネルギー汎関数を最小化することで得ることができて、
$$
V_{\rm eff}(r) = V_{\rm ext}(r) + \frac{e}{4 \pi \epsilon_0} \int dr' \frac{n(r')}{|r-r'|} + \frac{\delta E_{xc}}{\delta n(r)}
$$
となる。   
ここで重要なことは、上の方程式（コーンシャム方程式）と有効ポテンシャルの表式は**多体系における厳密な表式**ということである。つまり、有効ポテンシャルさえわかっていれば、「相互作用のない方程式によって固体中の電子集団のふるまいがわかる」ということである。その意味で、物理分野ではこの方程式を用いた計算を「第一原理計算」と呼ぶ。

### 2.4 コーンシャム方程式とタイトバインディング模型
密度汎関数理論は相互作用のない方程式で多体系を解くことができるという意味で非常に興味深い。
しかし、「有効ポテンシャルさえわかっていれば」がポイントである。このポテンシャルには、交換・相関エネルギー$E_{xc}$電子密度で変分した項が入っている。この項が厳密にわかっていえれば、固体中の電子の多体効果は全て取り扱えることになる。ただし、この項は厳密にはわからない。このままでは、**一休さんの「屏風のトラを出してください。そうすれば捕まえられます」**のようなものである。
実際の計算では、この項に近似を導入して計算することになる。近似には様々なものがあり、どのような近似を行うかで計算の精度が決まる。このような近似のせいで少数分子などの系では精度がよくないことが多く、分子を扱う量子化学の分野では「密度汎関数理論は平均場近似の一種で精度がよくない」といわれることになる。物理では「第一原理」、化学では「平均場近似」と扱いが異なるという意味でも興味深い理論である。


シュレーディンガー方程式とコーンシャム方程式は非常によく似ており、コーンシャム方程式に対してタイトバインディング模型を構築することができる。こうして得られたタイトバインディング模型には、「固体の情報がすべて入っている」ことになる。タイトバインディング模型になってしまいさえすれば、あとは好きな物理量を計算すればよい。つまり「屏風のトラはすでに屏風の外にいる」のである。もちろん、誰かが屏風のトラを外に出しているのが前提であるが。

### 2.5 タイトバインディング模型の性質
以上から、タイトバインディング模型が第一原理計算から求められたとする。その模型の性質を調べてみよう。
まず、$Nx$個の原子をもつタイトバインディング模型は

In [1]:
function calc_HtbModel(Nx)
    mat_Htb = zeros(Float64,Nx,Nx)    
    t = 1.0
    for i =1:Nx
        for dx in -1:1
            j = i + dx
            j += ifelse(j > Nx,-Nx,0)
            j += ifelse(j < 1,Nx,0)
            
            if dx ==0
                mat_Htb[i,j] = 0.0
            elseif abs(dx) == 1   
                mat_Htb[i,j] = -t    
            end
        end
    end 
    
    return mat_Htb
end

calc_HtbModel (generic function with 1 method)

とすると、固有値は

In [9]:
using Plots
gr()
Nx = 100
mat_H = calc_HtbModel(Nx)
energy,mat_v = eig(mat_H)

plot(energy)

<?xml version="1.0" encoding="utf-8"?>
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 20 
 
 
 40 
 
 
 60 
 
 
 80 
 
 
 100 
 
 
 -1.5 
 
 
 -1.0 
 
 
 -0.5 
 
 
 0.0 
 
 
 0.5 
 
 
 1.0 
 
 
 1.5 
 
<polyline clip-path="url(#clip5302)" style="stroke:#009af9; stroke-width:1; stroke-opacity:1; fill:none" points="
 39.1436,375.813 44.61,375.454 50.0763,375.454 55.5427,374.378 61.0091,374.378 66.4755,372.59 71.9419,372.59 77.4082,370.096 82.8746,370.096 88.341,366.906 
 93.8074,366.906 99.2737,363.033 104.74,363.033 110.206,358.492 115.673,358.492 121.139,353.301 126.606,353.301 132.072,347.481 137.538,347.481 143.005,341.054 
 148.471,341.054 153.938,334.047 159.404,334.047 164.87,326.485 170.337,326.485 175.803,318.401 181.269,318.401 186.736,309.824 192.202,309.824 197.669,300.79 
 203.135,300.79 208.601,291.333 214.068,291.333 219.534,281.492 225,281.492 230.467,271.305 235.933,271.305 241.4,260.811 246.866,260.811 252.332,250.054 
 257.799,250.054 263.265,239.074 268.731,239.074 274.198,227.916 279.664,227.916 285.131,216.623 290.597,216.623 296.063,205.24 301.53,205.24 306.996,193.812 
 312.462,193.812 317.929,182.384 323.395,182.384 328.862,171.001 334.328,171.001 339.794,159.709 345.261,159.709 350.727,148.55 356.193,148.55 361.66,137.571 
 367.126,137.571 372.593,126.813 378.059,126.813 383.525,116.32 388.992,116.32 394.458,106.132 399.925,106.132 405.391,96.2911 410.857,96.2911 416.324,86.8346 
 421.79,86.8346 427.256,77.8003 432.723,77.8003 438.189,69.2238 443.656,69.2238 449.122,61.1391 454.588,61.1391 460.055,53.5779 465.521,53.5779 470.987,46.5702 
 476.454,46.5702 481.92,40.1435 487.387,40.1435 492.853,34.3234 498.319,34.3234 503.786,29.1326 509.252,29.1326 514.718,24.5918 520.185,24.5918 525.651,20.7188 
 531.118,20.7188 536.584,17.5289 542.05,17.5289 547.517,15.0348 552.983,15.0348 558.449,13.2462 563.916,13.2462 569.382,12.1702 574.849,12.1702 580.315,11.811 
 
 "/>
 
 
 
 
 y1

Nx個の原子の中で、$N$個の電子がある場合には、この固有値を下から$N$個足したものが基底状態のエネルギーである。
ここで、タイトバインディング模型では、Nx個の電子の束縛状態を考えていることを思い出そう。もしNxを非常に大きくしていった場合、当然電子数$N$も非常に大きくなる。つまり、この二つの比$N/N_x$が意味のある量となる。この量をフィリングと呼ぶ。

この模型のフィリングが1/2のとき、つまりハーフフィリングの時、一粒子状態のなかで一番大きなエネルギーは、

In [12]:
println(energy[div(Nx,2)])

0.0


であり、ちょうどゼロである。
次に、粒子数がわからないとする。一粒子のエネルギーは-2から+2まである。このとき、これらの粒子のエネルギーを足し合わせて、一番低いエネルギーを探そうとすると、すべての負のエネルギーを足し合わせればよい。つまり、上の模型では、ハーフフィリングが一番多体系のエネルギーを最小とする。  
ほかのフィリングにするにはどうすればいいだろうか？　これは、エネルギーの原点をずらして、負のエネルギーの粒子の数を減らせばよいことになる。つまり、

In [13]:
function calc_HtbModel(Nx,μ)
    mat_Htb = zeros(Float64,Nx,Nx)    
    t = 1.0
    for i =1:Nx
        for dx in -1:1
            j = i + dx
            j += ifelse(j > Nx,-Nx,0)
            j += ifelse(j < 1,Nx,0)
            
            if dx ==0
                mat_Htb[i,j] = -μ
            elseif abs(dx) == 1   
                mat_Htb[i,j] = -t    
            end
        end
    end 
    
    return mat_Htb
end

calc_HtbModel (generic function with 2 methods)

として、固有値を求めて、ある$\mu$の時の負のエネルギーの数を数えてNxで割れば最小のエネルギーを持つフィリングが得られる。
$\mu = -1.5$のときには、

In [14]:
using Plots
gr()
Nx = 100
μ = -1.5
mat_H = calc_HtbModel(Nx,μ)
energy,mat_v = eig(mat_H)

count = 0
for ene in energy
    if ene > 0
        break
    end
    count += 1
end

println("Filling = ",count/Nx)

Filling = 0.23


フィリングは0.23となる。この$\mu$を「化学ポテンシャル」と呼ぶ。粒子の数を指定せずに、化学ポテンシャルを通じてフィリングを決めているので、これはグランドカノニカルアンサンブルを使っていることになる。この化学ポテンシャルは、基底状態においては低いエネルギーから順に粒子を詰めたときの一番高いエネルギーと等しく、このエネルギーは「フェルミエネルギー」とも呼ばれる。